doc/classes/TileMapLayer.xml

```diff
@@ -0,0 +1,308 @@
+		</member>
+		<member name="enabled" type="bool" setter="set_enabled" getter="is_enabled" default="true">
+			If [code]false[/code], disables completely this TileMapLayer (rendering, collision, navigation, scene tiles, etc.)
+		</member>
+		<member name="navigation_enabled" type="bool" setter="set_navigation_enabled" getter="is_navigation_enabled" default="true">
+			Enable or disable navigation regions.
+		</member>
+		<member name="navigation_visibility_mode" type="int" setter="set_navigation_visibility_mode" getter="get_navigation_visibility_mode" enum="TileMapLayer.VisibilityMode" default="0">
+			Show or hide the TileMapLayer's navigation meshes. If set to [constant VISIBILITY_MODE_DEFAULT], this depends on the show navigation debug settings.
+		</member>
+		<member name="rendering_quadrant_size" type="int" setter="set_rendering_quadrant_size" getter="get_rendering_quadrant_size" default="16">
+			The TileMapLayer's quadrant size. A quadrant is a group of tiles to be drawn together on a single canvas item, for optimization purposes. [member rendering_quadrant_size] defines the length of a square's side, in the map's coordinate system, that forms the quadrant. Thus, the default quandrant size groups together [code]16 * 16 = 256[/code] tiles.
+			The quadrant size does not apply on a Y-sorted TileMapLayer, as tiles are be grouped by Y position instead in that case.
+			[b]Note:[/b] As quadrants are created according to the map's coordinate system, the quadrant's "square shape" might not look like square in the TileMapLayer's local coordinate system.
+		</member>
+		<member name="tile_map_data" type="PackedByteArray" setter="set_tile_map_data_from_array" getter="get_tile_map_data_as_array" default="PackedByteArray(0, 0)">
+			The raw tile map data as a byte array.
+		</member>
+		<member name="tile_set" type="TileSet" setter="set_tile_set" getter="get_tile_set">
+			This TileMapLayer [TileSet].
+		</member>
+		<member name="use_kinematic_bodies" type="bool" setter="set_use_kinematic_bodies" getter="is_using_kinematic_bodies" default="false">
+			If [code]true[/code], this TileMapLayer collisions shapes will be instantiated as kinematic bodies. This can be needed for moving TileMapLayer nodes (i.e. moving platforms).
+		</member>
+		<member name="y_sort_origin" type="int" setter="set_y_sort_origin" getter="get_y_sort_origin" default="0">
+			This Y-sort origin value is added to each tile's Y-sort origin value. This allows, for example, to fake a different height level. This can be useful for top-down view games.
+		</member>
+	</members>
+	<signals>
+		<signal name="changed">
+			<description>
+				Emitted when this TileMapLayer's properties changes. This includes modified cells, properties or changes made to its assigned [TileSet].
+			</description>
+		</signal>
+	</signals>
+	<constants>
+		<constant name="VISIBILITY_MODE_DEFAULT" value="0" enum="VisibilityMode">
+			Use the debug settings to determine visibility.
```

Comment:
This enum could use some clarification/renaming. It sounds like it applies for the layer, while it only affects collision and navigation shapes. Maybe it could be called DebugVisibilityMode, with the description explicitly saying what it affects?

Reply:
I'll rename it to `DebugVisibilityMode`, but I am not sure what I can add to the values description. 
It's already detailed a bit more in the  `collision_visibility_mode` and `navigation_visibility_mode` properties respective descriptions.

Reply:
Then you could maybe reference them in the description?

Reply:
Something like "this enum is used by both `collision_visibility_mode` and `navigation_visibility_mode`" ?

I am not really sure I understand or if it makes sense. Those descriptions are used for each value, they are not really here to describe the enum itself, but instead each of its possible values. It does not seem the appropriate place to add more details.

Reply:
But those are not properties, they are enum values! They cannot be considered independently of properties that uses them. Once you know it's used by `collision_visibility_mode` and `navigation_visibility_mode`, it's pretty clear what "always hide" or "always show" mean.

I don't mind much repeating the information there, just for the sake of it, but I find that pretty verbose. I don't see why you would need to know more about those values if you don't know about the related properties in a first place.

But anyway, I don't mind much either way, to I'll repeat the context there then.